examples/platforms/simulation/radio.c

```diff
@@ -436,11 +436,7 @@ void platformRadioInit(void)
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
 static uint16_t getCslPhase(void)
 {
-    uint32_t curTime       = otPlatAlarmMicroGetNow();
-    uint32_t cslPeriodInUs = sCslPeriod * OT_US_PER_TEN_SYMBOLS;
-    uint32_t diff = ((sCslSampleTime % cslPeriodInUs) - (curTime % cslPeriodInUs) + cslPeriodInUs) % cslPeriodInUs;
-
-    return (uint16_t)(diff / OT_US_PER_TEN_SYMBOLS);
+    return (uint16_t)((sCslSampleTime - otPlatAlarmMicroGetNow()) / OT_US_PER_TEN_SYMBOLS);
```

Comment:
If the difference is larger than 'sCslPeriod', then 'phase' is larger than one period. That seems incorrect?

Reply:
This should be addressed when updating the CslSampleTime in upper layer.

Reply:
With current implementation in upper layer, I think the upper layer do ensure the diff is less than a period. (Temporarily I cannot think of any corner case). 

But I still think the assumption is a risk here. Can we add a log warn or assert here if the diff is larger than one period? Or we modulo the diff by the period. Then we don't need to have concern on those things. (No matter when is `getCslPhase` called and how many periods between now and sample time, the phase value we obtained is always correct). Thoughts?